src/ffi/index.ts

```diff
@@ -10,20 +10,40 @@ export const PACT_FFI_VERSION = '0.3.19';
 let ffi: typeof ffiLib;
 let ffiLogLevel: LogLevel;
 
-const initialiseFfi = (logLevel: LogLevel): typeof ffi => {
+const initialiseFfi = (logLevel: LogLevel, logFile?: string): typeof ffi => {
   logger.debug(`Initalising native core at log level '${logLevel}'`);
   ffiLogLevel = logLevel;
-  ffiLib.pactffiInitWithLogLevel(logLevel);
+  if (logFile) {
+    const convertLogLevelToFfiLoglevelFilter = {
+      trace: FfiLogLevelFilter.LOG_LEVEL_TRACE,
+      info: FfiLogLevelFilter.LOG_LEVEL_INFO,
+      debug: FfiLogLevelFilter.LOG_LEVEL_DEBUG,
+      warn: FfiLogLevelFilter.LOG_LEVEL_WARN,
+      error: FfiLogLevelFilter.LOG_LEVEL_ERROR,
+    };
+
+    logger.debug(
+      `Setting log file to '${convertLogLevelToFfiLoglevelFilter[logLevel]}'`
+    );
+    const res = ffiLib.pactffiLogToFile(
+      logFile,
+      convertLogLevelToFfiLoglevelFilter[logLevel]
```

Comment:
What happens if `convertLogLevelToFfiLoglevelFilter[logLevel]` is undefined?

Reply:
I think this whole inlined `convertLogLevelToFfiLoglevelFilter` constant would be better replaced with a `switch` - it would introduce no new concepts, and would give you error handling for free.

In general, if you need a really long name for something then it's a good signal that there's an alternative design that would be better.

Reply:
bad things probably 😂 or possibly, I don't know. Makes sense to test all the cases. I just did the bare minimum I could to get logging to file into pact-js when I was in the middle of diving into learning about FFI interfaces, so it is most certainly not the correct way, nor finished, I should move it to a draft and then it can be picked up either by me, or others if they feel inclined

Reply:
If it is undefined, napi will throw an error as we are expecting a numerical argument

```
Napi::Value PactffiLogToFile(const Napi::CallbackInfo& info) {
   // return: int
   Napi::Env env = info.Env();

  if (info.Length() < 2) {
    throw Napi::Error::New(env, "PactffiLogToFile(envVar) received < 2 arguments");
  }

    if (!info[0].IsString()) {
    throw Napi::Error::New(env, "PactffiLogToFile(envVar) expected a string");
  }

  if (!info[1].IsNumber()) {
    throw Napi::Error::New(env, "PactffiLogToFile(envVar) expected a number");
  }
```

I've also retained the existing `FfiLogLevelFilter[logLevel]` rather than needing to add a new function `convertLogLevelToFfiLoglevelFilter`